### PR TITLE
789 delete super event all subevents

### LINKED ISSUE #789: Deleting super-event should delete all sub-events whether they are pending or not.

### PR DESCRIPTION
**Pull request by minump**
## Description
Deleting a super events deletes sub events too. All subevents, whether it is published or not is deleted.

**Fixes #789**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: utilities/user_utilities.py [hunk 1]
```diff
@@ -1012,23 +1012,31 @@ def imagedId_from_eventId(eventId):
         return False
 
 
-def update_super_event_id(sub_event_id, super_event_id):
+def update_super_event_id(sub_event, sub_event_id, super_event_id, action):
```
[the remark this answers is not in the document]
Comment (minump) [reply utilities/user_utilities.py:1015]: addToSet is used for action=add and pull or unSet is used for action==pull. can have 2 separate functions for this.

**Comment by sandeep-ps** [comment]
@minump, please fix the merge conflicts on this PR when you get a chance. Thanks.

**Comment by minump** [comment]
converting to draft PR as more information is needed on whether events can be linked to multiple superevents.

**Comment by bingzhang** [comment]
@minump could you fix the conflicts and make it ready for review. thanks.

**Comment by sandeep-ps** [comment]
To add more details, we can currently proceed with the idea that one sub-event can be part of only one super event. Please make necessary changes to this PR and fix merge conflicts to proceed with the review. Thanks.

**Comment by minump** [comment]
code updated. ready for review.

File: user_events.py [hunk 2]
```diff
@@ -879,4 +889,4 @@ def sub_event(eventId):
     except Exception as ex:
         __logger.exception(ex)
         __logger.error("Redirect for eventid {} failed".format(eventId))
-        abort(500)
+        abort(500)
```
Comment (bingzhang) [inline user_events.py:892]: a new empty line after this line

File: user_events.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def user_events():
         session['select_status'] = select_status
 
     if request.method == 'POST':
-		#format : 'eventId=1234' /'category=Academic'/'eventId=1234&category=Academic'
+		# format : 'eventId=1234' /'category=Academic'/'eventId=1234&category=Academic'
```
Comment (bingzhang) [inline user_events.py:68]: this can be deleted?

**Comment by bingzhang** [comment]
@minump maybe just small things to be fixed. then we are good to merge it. thanks.

File: user_events.py (hunk 2 above)
Comment (minump) [reply user_events.py]: rectified

**Comment by minump** [comment]
> @minump maybe just small things to be fixed. then we are good to merge it. thanks.

modified code as per comments.

**Review by bingzhang — APPROVED** [review]

**Pull request merged**